test/tests/functional/pbs_qrun.py

```diff
@@ -100,30 +100,46 @@ def test_qrun_hangs(self):
         """
         node = self.mom.shortname
         time1 = int(time.time())
+        walltime = 1
+        self.server.manager(MGR_CMD_SET, SCHED,
+                            {'scheduling': 'False'})
+        err_msg = 'qrun: Not enough free nodes available'
         while True:
             self.server.manager(MGR_CMD_SET, NODE,
                                 {'resources_available.ncpus': 1}, id=node)
+
             j1 = Job(TEST_USER)
             j1.create_script('/bin/date')
+            a = {'Resource_List.walltime': walltime}
+            j1.set_attributes(a)
             self.server.submit(j1)
+
+            walltime = walltime + 1
+            a = {'Resource_List.walltime': walltime}
+            j1.set_attributes(a)
             jid2 = self.server.submit(j1)
+            walltime = walltime + 1
```

Comment:
Two things:
1) just having two jobs in the system in different equivalence classes won't slow things down.  If you had 500 or 1000 equivalence classes, that'd slow things down.
2) You don't delete the previous jobs, so the number of jobs will increase over time.  This does increase the number of equivalence classes, but not in a major way.  After 60 times, you'll have 120 jobs in the system.

If what you are trying to achieve is to have the scheduler in cycle when you do your qrun, then submit 500-1000 different jobs and run a cycle.  This cycle should take several seconds.  Do your qrun and see if it hangs.